lib/Routing/RedirectHandler.php

```diff
@@ -174,25 +174,42 @@ protected function buildRedirectResponse(Redirect $redirect, Request $request, $
             $url = replace_pcre_backreferences($url, $matches);
         }
 
-        if ($redirect->getTargetSite() && !preg_match('@http(s)?://@i', $url)) {
-            if ($targetSite = Site::getById($redirect->getTargetSite())) {
-                // if the target site is specified and and the target-path is starting at root (not absolute to site)
-                // the root-path will be replaced so that the page can be shown
-                $url = preg_replace('@^' . $targetSite->getRootPath() . '/@', '/', $url);
-                $url = $request->getScheme() . '://' . $targetSite->getMainDomain() . $url;
+        if(!preg_match('@http(s)?://@i', $url)) {
+            if ($redirect->getTargetSite()) {
+                if ($targetSite = Site::getById($redirect->getTargetSite())) {
+                    // if the target site is specified and and the target-path is starting at root (not absolute to site)
+                    // the root-path will be replaced so that the page can be shown
+                    $url = preg_replace('@^' . $targetSite->getRootPath() . '/@', '/', $url);
+                    $url = $request->getScheme() . '://' . $targetSite->getMainDomain() . $url;
+                } else {
+                    $this->logger->error('Site with ID {targetSite} not found', [
+                        'redirect' => $redirect->getId(),
+                        'targetSite' => $redirect->getTargetSite()
+                    ]);
+
+                    return null;
+                }
             } else {
-                $this->logger->error('Site with ID {targetSite} not found', [
-                    'redirect' => $redirect->getId(),
-                    'targetSite' => $redirect->getTargetSite()
-                ]);
+                $redirectDomain = $this->config['general']['domain'];
+                $sites = new Site\Listing();
```

Comment:
I stumbled upon a method to get the site by domain host and i think we utilize it here, so something like this: https://github.com/pimcore/pimcore/blob/2c01ab9c60fd222063cbaf59b290f812fa7fcaa8/bundles/CoreBundle/EventListener/Frontend/FrontendRoutingListener.php#L167

It leads better performance when you have a lot of sites. @BlackbitNeueMedien  WDYT?

Reply:
Hmm, does `Site::getByDomain()` really work? https://github.com/pimcore/pimcore/blob/004e782803f69a85c5d12bbf68bb932b70b4a747/models/Site/Dao.php#L60-L96

Currently the domains for a site are saved as serialized PHP array, not with the %-encoding which is for example used for relational data object fields.
<img width="548" alt="Bildschirmfoto 2020-04-20 um 14 21 54" src="https://user-images.githubusercontent.com/8749138/79751110-53fd9880-8312-11ea-96ad-2d7548236408.png">
So the SQL condition does not work and the code in the foreach loops does only regard wildcard domains.

Do we have to fix `Site\Dao::getByDomain()` or have I overseen something and it actually does work?

Reply:
I don't see any problem. It is using a LIKE operator so should work for additional domains also
![image](https://user-images.githubusercontent.com/5137917/79759536-16ebd300-831f-11ea-9c31-9349e3d0944c.png)

![image](https://user-images.githubusercontent.com/5137917/79759827-76e27980-831f-11ea-8328-4bcf7d57ff58.png)

Reply:
Yes, you are right, my fault - changed tis PR.